src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.GeneralizedTime.cs

```diff
@@ -216,20 +203,20 @@ public DateTimeOffset ReadGeneralizedTime(Asn1Tag expectedTag, bool disallowFrac
                     // T-REC-X.690-201510 sec 11.7.4
                     if (strict)
                     {
-                        throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                        throw new AsnContentException(SR.ContentException_InvalidUnderCerOrDer_TryBer);
                     }
                 }
                 else
                 {
                     Debug.Fail($"Unhandled value '{octet:X2}' in {nameof(FracState)}");
-                    throw new CryptographicException();
+                    throw new AsnContentException();
```

Comment:
Why both the Debug.Fail and the throw?

Reply:
Yeah, I don't think this `Debug.Fail` should be here. In general, invalid data from the client is "expected" in that we intend for it to go through graceful exception handling code paths. It doesn't violate invariants in our code.

Reply:
The fail is appropriate in that the only things which should transition into FracState are seeing a . or a , and both were already handled (an invalid character would have been processed at the point it decided if it was going to the Frac or Offset or Z states).  This could be folded with the comma case as an else (as an assert that it's a comma) if there's a strong desire to do so.